services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterTest.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    @Test
+    public void destroyImageThumbnailDerivative() throws Exception {
+        createContext(DESTROY_IMAGE_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("image/png", Cdr.FileObject.getURI()));
+
+        verify(destroySmallThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyImageThumbnailDerivativeCollection() throws Exception {
+        createContext(DESTROY_IMAGE_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("image/*", Cdr.Collection.getURI()));
+
+        verify(destroySmallThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyImageSrcCopyCollection() throws Exception {
+        createContext(DESTROY_SRC_COPY);
+
+        Map<String, Object> headers = createEvent("image/*", Cdr.Collection.getURI());
+        headers.put("CollectionThumb", true);
+        template.sendBodyAndHeaders("", headers);
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    // See if any messages are routed for object with no mimetype
+    @Test
+    public void destroyImageThumbnailNoDerivativeCollection() throws Exception {
+        getMockEndpoint("mock:direct:image.derivatives.destroy").expectedMessageCount(0);
+        getMockEndpoint("mock:direct:fulltext.derivatives.destroy").expectedMessageCount(0);
+
+        createContext(DESTROY_DERIVATIVES_ROUTE);
+        template.sendBodyAndHeaders("", createEvent("", Cdr.Collection.getURI()));
+        verify(binaryInfoProcessor).process(any(Exchange.class));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void destroyImageAccessDerivative() throws Exception {
+        createContext(DESTROY_ACCESS_COPY_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("image/png", Cdr.FileObject.getURI()));
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
```

Comment:
Similar to the collection thumbnail case, I'd say generally for sub-route tests like this you don't need to verify processors that aren't associated with that route